### PR TITLE
Vue Grid page implementation

### PR DESCRIPTION
**Pull request by SahilTara**
Summary of changes:
Implements the grid page for vue. You should now be able to generate a project with a grid page!

Should be reviewed after #647

Needs to be packaged locally to test until dll pr is in.
Related issues:
#199

### CODE REVIEW

**Review by Tanya0609 — COMMENTED** [review]
code looks good and I am able to generate a grid template with Vue. Will approve this PR once #631 is merged

**Review by jasmingoh — APPROVED** [review]
Was able to generate grid pages (Vue) and the code looks good!

**Comment by SahilTara** [comment]
closed in favour of #676

**Pull request closed without merging**
